### DISCUSSION TITLE
Can I keep a separate tab with primary selection items only?

### DISCUSSION THREAD

**Discussion by esauvisky** [hluk/CopyQ#2209] [Q&A]
Currently I don't use the "Store text selected using mouse" option because it severely pollutes my clipboard history. However, it would be extremely handy if I could keep a separate smaller tab with like the latest 50 items that ended up in the primary selection.

I read the API documentation and couldn't find any way of filtering these items with commands. Is this not possible?

> _What's the usecase?
> When developing software, copyq regex search functionality is extremely helpful to find previously used code, which really improves the overall workflow as it allows me to safely delete snippets and keep my code clean with the peace of mind of knowing that I can still recover those if need be. However, being a power user of the primary selection, I do not always explicitly copy stuff with ctrl+c, and frequently have to dig the original sources when I get no results on copyq.
> Furthermore, I wouldn't really mind (too much) to have "Store text selected using mouse" enabled and all the entries merged into the same tab, if it weren't for the fact that long entries with multiple lines (like 1000+) are very slow to filter out, and given the primary selection is usually full of garbage it really lags copyq._

**Comment by greg4b** [comment]
@esauvisky 
> Currently I don't use the "Store text selected using mouse" option because it severely pollutes my clipboard history. 

Please tell me how to activate the feature "Store text selected using mouse".

I'm very much interested in such type of feature.

Thanks!

**Comment by hluk** [comment]
The following command saves the mouse selections in a separate tab ([here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them)):

```ini
[Command]
Command="
    const selectionTab = 'Selections';

    var onClipboardChanged_ = onClipboardChanged;
    onClipboardChanged = function() {
        if (!isClipboard()) {
            setData(mimeOutputTab, selectionTab);
        }
        onClipboardChanged_();
    }"
Icon=\xe185
IsScript=true
Name=Tab for Primary Selection
```

**Reply by greg4b** [reply]
@hluk I've added your script - but mouse selections does not get into this tab. I'm on Win10, copyq 6.3.2 Thanks, Gregor

**Reply by hluk** [reply]
Windows do not support mouse selections. This would work only on Linux (X11 and sometimes Wayland).
